terragrunt/aws/api/lambda.tf

```diff
@@ -0,0 +1,20 @@
+module "api" {
+  source                   = "github.com/cds-snc/terraform-modules?ref=v0.0.44//lambda"
+  name                     = "api"
+  billing_tag_value        = var.billing_code
+  allow_api_gateway_invoke = true
+  api_gateway_source_arn   = aws_api_gateway_rest_api.api.arn
+  ecr_arn                  = aws_ecr_repository.api.arn
+  enable_lambda_insights   = true
+  image_uri                = "${aws_ecr_repository.api.repository_url}:latest"
+  vpc = {
+    security_group_ids = [module.rds.proxy_security_group_id, aws_security_group.api.id]
+    subnet_ids         = module.vpc.private_subnet_ids
+  }
+
+  policies = [
+    data.aws_iam_policy_document.api_policies.json,
+    data.aws_iam_policy_document.firehose_assume_role.json,
+    data.aws_iam_policy_document.write_waf_logs.json
```

Comment:
Instead of creating and attaching 3 policy documents, it's better if you consolidate them into one and attach that. 

Simliar to this policy document: https://github.com/cds-snc/scan-websites/blob/681b5a6e633dff7a45962ae94e73938deebc952c/terragrunt/aws/api/iam.tf#L28

The reason for this is that there is a limit to the number of IAM Polcies you can attach to a role. 

The limit is 20 policies so it's not like we are going to run into that limit anytime soon but it's just something to be aware of as a best practice. 

https://aws.amazon.com/premiumsupport/knowledge-center/iam-increase-policy-size/#:~:text=You%20can%20attach%20up%20to,to%20IAM%20roles%20and%20users.

This is non-blocking

Reply:
thanks! good to know about the IAM best practise. I removed the policy attachments since i didn't intend to attach those to the lambda! They were for the waf so it can write to kinesis